### ISSUE TITLE
DataTable on other Sheet: "Value cannot be null. (Parameter 'source')" if source table has no records

### ISSUE THREAD

**Issue by GitAleB** [ClosedXML/ClosedXML#2072]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x ] Yes (latest develop (2023-04-25)
- [] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

Version > 0.76
Tested with 0.97 and 0.101

**What is the current behavior?**

Excel with two Sheets:
A) Pivot Table
B) Data Table
If Data Table does not hold any data -> Exception thrown:   
```
System.ArgumentNullException: Value cannot be null. (Parameter 'source')   
at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)  
at System.Linq.Enumerable.ToList[TSource](IEnumerable``1 source)   
at ClosedXML.Excel.PivotLabelFieldReference.Match(PivotTableInfo pti, IXLPivotTable pt)
```

**What is the expected behavior or new feature?**

Data Table has no records -> Excel can be saved

**Is this a regression from the previous version?**

Yes: No problem with Version 0.76

## Reproducibility
Use the attached Excel an run the code "Code to reproduce problem":

**Code to reproduce problem:**
```c#

    public static int Main(string[] args)
    {
        // Note: adjust BugPivotSourceOnOtherSheet to match project name
        // xlsx is attached: [Sample.xlsx](https://github.com/ClosedXML/ClosedXML/files/11325551/Sample.xlsx)
        using (Stream stream = typeof(Program).Assembly.GetManifestResourceStream("BugPivotSourceOnOtherSheet.Sample.xlsx")!)
        {

            using (XLWorkbook book = new XLWorkbook(stream))
            {
                // no further modification... save
                using (Stream fileStream = File.Open("Out.xlsx", FileMode.Create))
                {
                    // Upon SaveAs: Exception "Value cannot be null" is thrown
                    book.SaveAs(fileStream);
                }
            }
        }
        return 0;
    }  
```

- [ x] I attached a sample spreadsheet.  (You can drag files on to this issue)
- [Sample.xlsx](https://github.com/ClosedXML/ClosedXML/files/11325562/Sample.xlsx)

**Comment by GitAleB** [comment]
Addendum: A fix would be pretty easy:
Class `PivotLabelFieldReference.cs`: modify:
```
  //var values = pti.Fields[PivotField.SourceName].DistinctValues.ToList();
  var values = pti.Fields[PivotField.SourceName].DistinctValues?.ToList();
  if (values == null)
      return Array.Empty<Int32>();
```

If you want a PR: let me know

**Comment by jahav** [comment]
Hi, this should be fixed by #1238 (at least the sample didn't throw and file opened, I haven't look at it too closely). It has been in kept limbo for a while, but I cleaned it up and want to merge it as a part of 0.102.

**Comment by GitAleB** [comment]
> Hi, this should be fixed by #1238 (at least the sample didn't throw and file opened, I haven't look at it too closely). It has been in kept limbo for a while, but I cleaned it up and want to merge it as a part of 0.102.

Hi
The error I described did appear on SaveAs(stream). I just tested it on branch "pivot-sources", commit 74575cc. It seems to fix the issue. Great!
What is the expected release date of 0.102?

**Comment by jahav** [comment]
@GitAleB  Start of June, https://github.com/ClosedXML/ClosedXML/discussions/2067

**Comment by GitAleB** [comment]
> @GitAleB Start of June, #2067

👍 Perfect... thanks!

**Comment by jahav** [comment]
This has been fixed in 0.102, as part of #1238, I just forgot it. 

It has been a massive refactor o pivot tables, so I don't remember exact reason why it was fixed, but I believe that it was separation of field source names, field custom names and header texts in source data (well, that is basically source name unless there is a duplicate).

To be more precise, the exception in the sample fails on missing source name of a field is `ContractorName`, but custom name of a field is 'Auftragnehmer'.
